plugins/GSdx/GSTextureCache.h

```diff
@@ -105,6 +105,47 @@ class GSTextureCache
 		void Update();
 	};
 
+	class Palette
+	{
+	private:
+		uint32* m_clut; // Pointer to a copy of relevant clut
+		GSTexture* m_tex_palette; // Pointer to valid texture with relevant clut as content
+
+	public:
+		Palette(); // Default constructor
+		Palette(const Palette& palette); // Copy constructor
+		Palette(const GSRenderer* renderer, uint16 pal); // Creates a copy of the current clut and a texture with its content
+
+		// Getter for clut pointer
+		__forceinline uint32* GetClut();
+
+		// Getter for palette texture pointer
+		__forceinline GSTexture* GetPaletteGSTexture();
+	};
+
+	class PaletteMap
+	{
+	private:
+		static const uint16 MAX_SIZE = 2048; // Max size of the map
+		uint16 m_size = 0; // Both size counter and valid stack top index
+		std::array<Palette, MAX_SIZE> m_palettes_stack; // Stack of Palette to be used to avoid memory fragmentation
+		
+		// Array of 2 multimaps, the first for 64B palettes and the second for 1024B palettes.
+		//     Each multimap has the clut hash value as key and the relevant palettes as values (multiple values possible for each key)
```

Comment:
Small nitpick, extra spaces here.

Reply:
Oh I put those by purpose, but yes maybe it's better without them!